crates/swc_ecma_codegen/src/lib.rs

```diff
@@ -820,11 +824,7 @@ where
                 }
             }
             OptChainBase::Call(ref e) => {
-                if let Expr::New(new) = &*e.callee {
-                    self.emit_new(new, false)?;
-                } else {
-                    emit!(e.callee);
-                }
+                emit!(e.callee);
```

Comment:
It's possible to write `new Date()?.toLocaleDateString()`. 
But `new Date?.()` is invalid and will be caught by the parse/lint stage.

Reply:
I think we can use `debug_assert` in such cases, we have the same situation in CSS codege - in theory developer can modify AST and create invalid code